modules/nlu/src/backend/predict-pipeline.ts

```diff
@@ -144,6 +144,23 @@ async function extractEntities(input: PredictStep, predictors: Predictors, tools
     }
   )
 
+  if (alternateUtterance) {
+    const sysEntities = (await tools.duckling.extract(alternateUtterance.toString(), utterance.languageCode)).map(
+      mapE1toE2Entity
+    )
+
+    _.forEach(
+      [
+        ...extractListEntities(alternateUtterance, predictors.list_entities),
+        ...extractPatternEntities(alternateUtterance, predictors.pattern_entities),
+        ...sysEntities
+      ],
+      entityRes => {
+        input.alternateUtterance.tagEntity(_.omit(entityRes, ['start, end']), entityRes.start, entityRes.end)
+      }
+    )
+  }
```

Comment:
Instead of duplicating, can you make this a method and pass it an argument ? (utterance/alternateutterance

Reply:
Will come in another pr with all of alternateUtterance predictions duplicate